### PR TITLE
Display normalized values in spectrum viewer image view

### LINKED ISSUE #2456: Show normalised image in spectrum view
In the spectrum viewer, on the image view, we currently always show the sample data (averaged over tof bins), even when normalisation is enabled.

Instead we should show the normalised values.

### PR DESCRIPTION
**Pull request by ashleymeigh2**
### Description

- Updated spectrum viewer to display normalized values in the image view when normalization is enabled.
- Added `get_normalized_averaged_image()` to the model for normalized image data.
- Updated presenter to fetch and display the correct image based on normalization state.

### Developer Testing 

- Verified unit tests pass: `python -m pytest -vs`
- Tested normalized and non-normalized image rendering in the spectrum viewer.

### Acceptance Criteria

- [x] Unit tests pass: `python -m pytest -vs`
- [x] Normalized values display when normalization is enabled.
- [x] Non-normalized values display when normalization is disabled.
- [x] Edge cases (e.g., missing stacks) handled correctly.

### CODE REVIEW

File: mantidimaging/gui/windows/spectrum_viewer/model.py [hunk 1]
```diff
@@ -130,6 +130,18 @@ def set_stack(self, stack: ImageStack | None) -> None:
...
+        """
+        Get the normalized averaged image if both sample and normalization stacks are available.
+        """
+        if self._stack is None or self._normalise_stack is None:
+            return None
+
+        tof_slice = slice(self.tof_range[0], self.tof_range[1] + 1)
+        sample_data = self._stack.data[tof_slice].mean(axis=0)
+        norm_data = self._normalise_stack.data[tof_slice].mean(axis=0)
+        return np.divide(sample_data, norm_data, out=np.zeros_like(sample_data), where=norm_data != 0)
```
Comment (samtygier-stfc) [inline mantidimaging/gui/windows/spectrum_viewer/model.py:143]: Can you also add the shutter count correction. See the last few lines of `get_spectrum()`.

**Review by samtygier-stfc — CHANGES_REQUESTED** [review]
I think the test is failing because `self.view.normalisation_enabled()` will be true by default in the mocked view. Might fix it to add `self.view.normalisation_enabled.return_value = False` in to the test.

**Comment by samtygier-stfc** [comment]
And it would be good to have a test for the new function in `model_test.py`

**Comment by samtygier-stfc** [comment]
Looks like this still needs a yapf fix

File: mantidimaging/gui/windows/spectrum_viewer/presenter.py [hunk 2]
```diff
@@ -169,9 +169,12 @@ def show_new_sample(self) -> None:
         Show the new sample in the view and update the spectrum and
         image view accordingly. Resets the ROIs.
         """
-
-        averaged_image = self.model.get_averaged_image()
+        if self.view.normalisation_enabled():
+            averaged_image = self.model.get_normalized_averaged_image()
+        else:
+            averaged_image = self.model.get_averaged_image()
```
Comment (samtygier-stfc) [inline mantidimaging/gui/windows/spectrum_viewer/presenter.py:175]: This also needs to be done in `handle_range_slide_moved()`

Also redraw will be needed in `handle_enable_normalised()` to that the image changes with the normalise box is ticked or unticked.

**Review by samtygier-stfc — CHANGES_REQUESTED** [review]

**Review by samtygier-stfc — CHANGES_REQUESTED** [review]
Now that there is similar code in 3 places, I think it is worth pulling
```python
        averaged_image = (self.model.get_normalized_averaged_image()
                          if self.view.normalisation_enabled() else self.model.get_averaged_image())
        if averaged_image is None:
            image_shape = self.model.get_image_shape()
            averaged_image = np.zeros(image_shape, dtype=np.float32)
        self.view.set_image(averaged_image, autoLevels=False)
```

into its own method.

File: mantidimaging/gui/windows/spectrum_viewer/presenter.py [hunk 3]
```diff
@@ -305,6 +312,11 @@ def handle_enable_normalised(self, enabled: bool) -> None:
             self.spectrum_mode = SpecType.SAMPLE
         self.redraw_all_rois()
         self.view.display_normalise_error()
+        averaged_image = self.model.get_normalized_averaged_image() if enabled else self.model.get_averaged_image()
+        if averaged_image is None:
+            image_shape = self.model.get_image_shape()
+            averaged_image = np.zeros(image_shape, dtype=np.float32)
+        self.view.set_image(averaged_image, autoLevels=True)
```
Comment (samtygier-stfc) [inline mantidimaging/gui/windows/spectrum_viewer/presenter.py:319]: Can this be replaced with a call to `self.update_displayed_image(autoLevels=True)`

File: mantidimaging/gui/windows/spectrum_viewer/presenter.py [hunk 4]
```diff
@@ -232,6 +238,7 @@ def redraw_all_rois(self) -> None:
             widget_roi = self.view.spectrum_widget.get_roi(roi_name)
             spectrum = self.model.get_spectrum(widget_roi, self.spectrum_mode, self.view.shuttercount_norm_enabled())
             self.view.set_spectrum(roi_name, spectrum)
+        self.update_displayed_image(autoLevels=False)
```
Comment (samtygier-stfc) [inline mantidimaging/gui/windows/spectrum_viewer/presenter.py:241]: Does this need calling here, it does not look like the old code did an image redraw here.

**Review by samtygier-stfc — CHANGES_REQUESTED** [review]

**Review by samtygier-stfc — APPROVED** [review]
Looks good now. Thanks

**Review by samtygier-stfc — APPROVED** [review]
